test/Mono.Linker.Tests.Cases/Reflection/TypeHierarchyReflectionWarnings.cs

```diff
@@ -0,0 +1,444 @@
+			[Kept]
+			[KeptBackingField]
+			[KeptAttributeAttribute (typeof (DynamicallyAccessedMembersAttribute))]
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicMethods)]
+			public static string DAMTProperty {
+				// Property access reports warnings on getter/setter
+				[ExpectedWarning ("IL2114", nameof (AnnotatedPublicProperties), nameof (DAMTProperty) + ".get")]
+				get;
+			}
+		}
+
+		[Kept]
+		[KeptAttributeAttribute (typeof (DynamicallyAccessedMembersAttribute))]
+		[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicEvents)]
+		class AnnotatedPublicEvents
+		{
+			[Kept]
+			[KeptMember (".ctor(System.Object,System.IntPtr)")]
+			[KeptMember ("Invoke(System.Object,System.Int32)")]
+			[KeptBaseType (typeof (MulticastDelegate))]
+			public delegate void MyEventHandler (object sender, int i);
+
+			[Kept]
+			// We always keep event methods when an event is kept, so this generates warnings
+			// on the event itself (since an event access is considered to reference the annotated add method),
+			// and on the add method (if it is accessed through reflection).
+			[ExpectedWarning ("IL2026", "--RUC on add_RUCEvent--")]
+			public event MyEventHandler RUCEvent {
+				[Kept]
+				[ExpectedWarning ("IL2112", nameof (AnnotatedPublicEvents), "--RUC on add_RUCEvent--")]
+				[KeptAttributeAttribute (typeof (RequiresUnreferencedCodeAttribute))]
+				[RequiresUnreferencedCode ("--RUC on add_RUCEvent--")]
+				add { }
+				[Kept]
+				remove { }
+			}
+		}
+
+		[Kept]
+		[KeptAttributeAttribute (typeof (DynamicallyAccessedMembersAttribute))]
+		[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicMethods)]
+		interface RequiredInterface
+		{
+			// Removed, because keeping the interface on its own
+			// doesn't apply its type annotations
+			[RequiresUnreferencedCode ("--RUC on RequiredInterface.UnusedMethod--")]
+			void RUCMethod ();
+		}
+
+		[Kept]
+		[KeptInterface (typeof (RequiredInterface))]
+		[KeptAttributeAttribute (typeof (DynamicallyAccessedMembersAttribute))]
+		[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.Interfaces)]
+		class AnnotatedInterfaces : RequiredInterface
+		{
+			[Kept]
+			[KeptAttributeAttribute (typeof (RequiresUnreferencedCodeAttribute))]
+			// This should produce a warning: https://github.com/mono/linker/issues/2161
+			[RequiresUnreferencedCode ("--RUC on AnnotatedInterfaces.UnusedMethod--")]
+			public void RUCMethod () { }
+		}
+
+		[KeptAttributeAttribute (typeof (DynamicallyAccessedMembersAttribute))]
+		[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicMethods)]
+		class AnnotatedBase
+		{
+			[Kept]
+			[KeptAttributeAttribute (typeof (RequiresUnreferencedCodeAttribute))]
+			[ExpectedWarning ("IL2112", "--RUC on AnnotatedBase--")]
+			[RequiresUnreferencedCode ("--RUC on AnnotatedBase--")]
+			public void RUCMethod () { }
+		}
+
+		[KeptBaseType (typeof (AnnotatedBase))]
+		[ExpectedWarning ("IL2113", "--RUC on AnnotatedBase--")]
```

Comment:
Ideally we would not produce this warning - there's nothing wrong the derived type did in this case - all the blame should be on the base type - which already generates a warning for this anyway.

I think it's OK to not fix this if it's hard, but logically it should not happen.

Reply:
(As in create an issue if it's too hard to fix)

Reply:
I'll leave that for a follow-up: https://github.com/mono/linker/issues/2175